controller/tenants_updates_blackbox_test.go

```diff
@@ -0,0 +1,341 @@
+			apptest.ShowTenantUpdatesUnauthorized(t, ctx, svc, ctrl, nil)
+		})
+
+		t.Run("unauthorized", func(t *testing.T) {
+			// given
+			ctx, identity, err := testauth.EmbedUserTokenInContext(context.Background(), testauth.NewIdentity())
+			require.NoError(t, err)
+			tk := goajwt.ContextJWT(ctx)
+			require.NotNil(t, tk)
+			authzHeader := fmt.Sprintf("Bearer %s", tk.Raw)
+			gock.New("https://test-tenant").
+				Get("/api/tenant/updates").
+				MatchHeader("Authorization", authzHeader).
+				Reply(http.StatusUnauthorized)
+			// when
+			apptest.ShowTenantUpdatesUnauthorized(t, ctx, svc, ctrl, &authzHeader)
+			// then check that an audit record was created
+			assertAuditLog(t, s.DB, *identity, auditlog.ShowTenantUpdate, auditlog.EventParams{})
+		})
+
+		t.Run("internal server error", func(t *testing.T) {
+			// given
+			ctx, identity, err := testauth.EmbedUserTokenInContext(context.Background(), testauth.NewIdentity())
+			require.NoError(t, err)
+			tk := goajwt.ContextJWT(ctx)
+			require.NotNil(t, tk)
+			authzHeader := fmt.Sprintf("Bearer %s", tk.Raw)
+			gock.New("https://test-tenant").
+				Get("/api/tenant/updates").
+				MatchHeader("Authorization", authzHeader).
+				Reply(http.StatusInternalServerError)
+			// when
+			apptest.ShowTenantUpdatesInternalServerError(t, ctx, svc, ctrl, &authzHeader)
+			// then check that an audit record was created
+			assertAuditLog(t, s.DB, *identity, auditlog.ShowTenantUpdate, auditlog.EventParams{})
+		})
+	})
+}
+func (s *TenantUpdatesControllerBlackboxTestSuite) TestStartTenantUpdates() {
+	// given
+	config := testconfig.NewTenantUpdatesControllerConfigurationMock(s.T())
+	config.GetTenantServiceURLFunc = func() string {
+		return "https://test-tenant"
+	}
+	svc, ctrl := newTenantUpdatesController(config, s.app)
+	defer gock.OffAll()
+
+	s.T().Run("ok", func(t *testing.T) {
+
+		t.Run("all clusters all envs", func(t *testing.T) {
+			// given
+			ctx, identity, err := testauth.EmbedUserTokenInContext(context.Background(), testauth.NewIdentity())
+			require.NoError(t, err)
+			tk := goajwt.ContextJWT(ctx)
+			require.NotNil(t, tk)
+			authzHeader := fmt.Sprintf("Bearer %s", tk.Raw)
+			gock.New("https://test-tenant").
+				Post("/api/tenant/updates").
+				MatchHeader("Authorization", authzHeader).
+				Reply(http.StatusAccepted).BodyString(`{"data":"whatever"}`)
+			// when
+			apptest.StartTenantUpdatesAccepted(t, ctx, svc, ctrl, nil, nil, &authzHeader)
+			// then check that an audit record was created
+			assertAuditLog(t, s.DB, *identity, auditlog.StartTenantUpdate, auditlog.EventParams{})
+		})
+
+		t.Run("single clusters single env", func(t *testing.T) {
+			// given
+			ctx, identity, err := testauth.EmbedUserTokenInContext(context.Background(), testauth.NewIdentity())
+			require.NoError(t, err)
+			tk := goajwt.ContextJWT(ctx)
+			require.NotNil(t, tk)
+			authzHeader := fmt.Sprintf("Bearer %s", tk.Raw)
+			gock.New("https://test-tenant").
+				Post("/api/tenant/updates").
```

Comment:
you could also match the params (cluster url and env) using the `MatchParam` method

Reply:
ok, good point, let me do that

Reply:
done in 02cc202